instrumentation/jetty-httpclient/jetty-httpclient-9.2/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/jetty/httpclient/v9_2/JettyHttpClient9Instrumentation.java

```diff
@@ -43,38 +45,50 @@ public void transform(TypeTransformer transformer) {
   @SuppressWarnings("unused")
   public static class JettyHttpClient9Advice {
 
+    public static class AdviceLocals {
+      public final Context context;
+      public final Scope scope;
+
+      public AdviceLocals(Context context, Scope scope) {
+        this.context = context;
+        this.scope = scope;
+      }
+    }
+
+    @AssignReturned.ToArguments(@ToArgument(value = 1, index = 1))
     @Advice.OnMethodEnter(suppress = Throwable.class)
-    public static void addTracingEnter(
+    public static Object[] addTracingEnter(
         @Advice.Argument(value = 0) HttpRequest httpRequest,
-        @Advice.Argument(value = 1, readOnly = false) List<Response.ResponseListener> listeners,
-        @Advice.Local("otelContext") Context context,
-        @Advice.Local("otelScope") Scope scope) {
+        @Advice.Argument(1) List<Response.ResponseListener> originalListeners) {
+
+      List<Response.ResponseListener> listeners = originalListeners;
```

Comment:
I think it would be easier to follow if you skip on declaring `listeners here` and instead at the end of the method do the following:

```
List<Response.ResponseListener> wrappedListeners = wrapResponseListeners(parentContext, listeners);
return new Object[] {new AdviceLocals(context, context.makeCurrent()), wrappedListeners};
```